pkg/controller/machine/machine.go

```diff
@@ -152,6 +153,45 @@ func NewMachineController(
 		name: name,
 	}
 
+	go func() {
+		metricGatheringExecutor := func() {
+			machines, err := machineLister.List(labels.NewSelector())
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("faild to list machines for SetInstanceNumberForMachines: %v", err))
+				return
+			}
+			var machineList clusterv1alpha1.MachineList
+			for _, machine := range machines {
+				machineList.Items = append(machineList.Items, *machine.DeepCopy())
+			}
+			if len(machineList.Items) < 1 {
+				return
+			}
+
+			// We assume that all machines are on the same provider
```

Comment:
is this assumption appropriate? The reason for doing it like that is

* Getting the provider for a machine is rather cumbersome
* We would otherwise need to maintain a list per provider, which makes the provider abstraction feel leaky, ideally the controller wouldn't have to know anything about what providers exist

Reply:
Done